src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -479,8 +483,34 @@ class ArnoldAttributes : public IECoreScenePreview::Renderer::AttributesInterfac
 
 		};
 
+		struct Curves
+		{
+
+			Curves( const IECore::CompoundObject *attributes )
+			{
+				mode = attributeValue<string>( g_curvesModeAttributeName, attributes, "ribbon" );
+				minPixelWidth = attributeValue<float>( g_curvesMinPixelWidthAttributeName, attributes, 0.0f );
+			}
+
+			IECore::InternedString mode;
+			float minPixelWidth;
+
+			void hash( IECore::MurmurHash &h ) const
+			{
+				h.append( mode );
+				h.append( minPixelWidth );
+			}
+
+			void apply( AtNode *node ) const
+			{
+				AiNodeSetStr( node, "mode", mode.c_str() );
```

Comment:
The oriented stuff will need doing on the Cortex side. I thought the thick mode was the one that was the reason for the urgency on this, so bashed that out without doing the Cortex bit. Want me to do the rest too? And before or after the OSLCode stuff?